cider-stacktrace.el

```
@@ -561,6 +561,26 @@ Achieved by destructively manipulating the `cider-stacktrace-suppressed-errors'
         (button-put button 'help-echo "Click to promote these stacktraces."))
       (button-put button 'suppressed (not suppressed)))))
 
+(defcustom cider-jdk-src-paths '("/usr/lib/jvm/openjdk-8/src.zip")
+  "Used by `cider-stacktrace-navigate'.
+Zip files work, but it's better to extract them and put the directory paths here.
+Clojure sources here: https://repo1.maven.org/maven2/org/clojure/clojure/1.8.0/."
+  :type '(list string))
+
+(defun cider-resolve-java-class (class)
+  "Return a zip path to a file that corresponds to CLASS."
+  (when class
+    (let ((file-name (concat (replace-regexp-in-string "\\." "/" class) ".java")))
+      (cl-find-if
+       'file-exists-p
+       (mapcar
+        (lambda (d)
+          (if (file-directory-p d)
+              (expand-file-name file-name d)
+            (when (file-exists-p d)
```

Perhaps we should raise an error if the file is not `zip` or `jar`?